sound/soc/sof/ipc4-topology.h

```diff
@@ -401,6 +401,24 @@ struct sof_ipc4_src {
 	struct sof_ipc4_msg msg;
 };
 
+/**
+ * struct sof_ipc4_asrc ASRC config data
+ * @base_config: IPC base config data
+ * @sink_rate: Output rate for sink module
+ * @asynchronous_mode: synchronous 0, asynchronous 1.
+ * @operation_mode: push 0, pull 1.
```

Comment:
what does this mean?

Reply:
In push mode the  input frames count is fixed and known, output frames count varies.

In pull mode the input frames count varies, output frames count is fixed and known.

For a follower-DAI the playback direction would be push, and capture direction pull.

Reply:
That was partly my question, 0 and 1 are just not self-explanatory.

But the other question is really that if the number of frames can vary, then how does the buffering model work? we define IBS/OBS for IPC4, so is there a need to allocate larger buffers to account for the variability of the ASRC?

The only other sort of processing of this nature would be a time-stretching algorithm, and I really can't reconcile how these sort of algorithms need to be described by our topology. Best for @ranj063 to chime in.

Reply:
at this time, there is an internal circular buffer of ASRC itself to store the input/output samples. and the buffer size is 10 frames bigger than a period samples.